reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt

```diff
@@ -330,6 +330,17 @@ class FreemarkerTemplateProcessor(
         @Suppress("UNUSED") // This function is used in the templates.
         fun filterForUnresolvedVulnerabilities(vulnerabilities: List<Vulnerability>): List<Vulnerability> =
                 vulnerabilities.filter { resolutionProvider.getVulnerabilityResolutionsFor(it).isEmpty() }
+
+        /**
+         * Return a human-readable string that is determined based on the given score and scoring system.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun getSeverityString(scoreSystem: String?, score: String) =
+            when (scoreSystem) {
+                "CVSS2" -> Vulnerability.Cvss2Rating.fromScore(score.toFloat())?.name?.toLowerCase()
+                "CVSS3" -> Vulnerability.Cvss3Rating.fromScore(score.toFloat())?.name?.toLowerCase()
+                else -> "unknown"
+            }
```

Comment:
How about moving this function to a `companion object` inside the `Vulnerability` class instead?

Reply:
Actually, I believe we should do some more small refactorings first. now that we have the `VulnerabilityReference` class:

1. Commit: Move the `Cvss2Rating` and `Cvss3Rating` *enums* from `Vulnerability` to `VulnerabilityReference`.
2. Commit: Add this function to a `companion object` in `VulnerabilityReference`.

Reply:
It might make sense to align these refactorings with requirements that come out of the work on #3503. Rule authors will need mechanisms to handle different scoring systems as well.

Reply:
Pushed my first draft of #3503 to this PR: https://github.com/oss-review-toolkit/ort/pull/3961

I designed the rules API in a way that the actual value from the `VulnerabilityReference` is being used and needs to be provided by the rules author.
So I don't think that this will be affected by this PR.